File: rb/features.rb
# WebsiteAnalysisApis2 SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module WebsiteAnalysisApis2Features
  def self.make_feature(name)
    case name
    when "base"
      WebsiteAnalysisApis2BaseFeature.new
    when "test"
      WebsiteAnalysisApis2TestFeature.new
    else
      WebsiteAnalysisApis2BaseFeature.new
    end
  end
end

File: py/websiteanalysisapis2_sdk/features.py
# WebsiteAnalysisApis2 SDK feature factory

from websiteanalysisapis2_sdk.feature.base_feature import WebsiteAnalysisApis2BaseFeature
from websiteanalysisapis2_sdk.feature.test_feature import WebsiteAnalysisApis2TestFeature


def _make_feature(name):
    features = {
        "base": lambda: WebsiteAnalysisApis2BaseFeature(),
        "test": lambda: WebsiteAnalysisApis2TestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
